src/dos/cdrom_image.cpp

```diff
@@ -303,44 +315,45 @@ bool CDROM_Interface_Image::GetAudioTracks(int& start_track_num, int& lead_out_n
 #endif
 		return false;
 	}
-	start_track_num = tracks.begin()->number;
-	track_const_iter lead_out(prev(tracks.end()));
-	lead_out_num = lead_out->number;
+	start_track_num = tracks.front().number;
+	track_const_iter lead_out = prev(tracks.end());
+	end_track_num = prev(lead_out)->number;
 	lead_out_msf = frames_to_msf(lead_out->start + 150);
```

Comment:
Suggestion:

    start_track_num = tracks.front().number;
    end_track_num = next(tracks.crbegin())->number;
    lead_out_msf = frames_to_msf(tracks.back().start + 150);

`crbegin` is const reverse iterator; but after looking at it, maybe simpler:

    end_track_num = tracks[tracks.size() - 2].number;

would be more readable?

Reply:
I like your first suggestion; it's a huge readability improvement over what I had. I like the next(crbegin) as well; even though it feels esoteric, it's growing on me now.  I also like the idea of not using indexes if you're getting fixed bounding elements; and reserving indexes for true random access.